protocol/mqtt_paho/v2/protocol.go

```diff
@@ -107,6 +106,16 @@ func (p *Protocol) Send(ctx context.Context, m binding.Message, transformers ...
 	return err
 }
 
+// publishMsg generate a new paho.Publish message from the p.publishOption
+func (p *Protocol) publishMsg() *paho.Publish {
+	return &paho.Publish{
+		QoS:        p.publishOption.QoS,
+		Retain:     p.publishOption.Retain,
+		Topic:      p.publishOption.Topic,
+		Properties: p.publishOption.Properties,
+	}
+}
```

Comment:
If I'm understanding correctly, the race condition is that the `publishOption` can be changed before the message is published?

Since there is only one place to set this field in the struct (https://github.com/cloudevents/sdk-go/blob/e6a74efbacbf4ac70fc8ed598e678f94b6be7a6d/protocol/mqtt_paho/v2/option.go#L29-L37), maybe it would make sense to make the copy there instead of every time a message is published?

Reply:
+1 

Left a comment in the issue to better understand the issue/root cause. What is causing the race? Why is `publishOption` changing during concurrent calls? Asking all these questions because I didn't write the implementation and not familiar with MQ. Furthermore, if we change as per the above, I wonder what the implication is for existing users because we're changing the semantics here (copy instead of pointer).

Reply:
Thanks @Cali0707!  The code block you referred to is only invoked once when initializing the `client`. The issues arose when sending messages in many goroutines. Here is the [detail](https://github.com/cloudevents/sdk-go/pull/1094#issue-2505143016)

Reply:
FYI @embano1.

I append the root cause [here](https://github.com/cloudevents/sdk-go/pull/1094#issue-2505143016). 

The `publishOption` is a pointer reference by the MQTT message, which is used to hold the CloudEvent payload. The message(`publishOption`) is shared across goroutines and will be changed by them.

If we change as per the above, the user should not perceive anything.

Reply:
> and will be changed by them

Can you please point me to the code/line where this happens?
Also, ideally you also write a test please to surface the current issue (panics) and how this PR fixes it, perhaps there's more issues due to the current implementation which we can detect this way 🙏

Reply:
@embano1 No worries!

Here’s the [code line](https://github.com/cloudevents/sdk-go/pull/1094/files#diff-c0aa4c2cb67e2b13ef708c2704bf681052f3c816e730574f94ff807cf8dd4b6bR97) where the panic occurs.

I’ve also written an [integration test](https://github.com/cloudevents/sdk-go/pull/1094/commits/74611564cee99d2bb35e7b532107c44df62140be) that covers the concurrent panic issue, which you can use to reproduce it.